Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs

```diff
@@ -313,7 +315,22 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
             var oldestByEvictability = contentHashesWithInfo.Take(contentHashesWithInfo.Count / 2).ApproximateSort(comparer, intoEffectiveLastAccessTimes, _configuration.EvictionPoolSize, _configuration.EvictionWindowSize, _configuration.EvictionRemovalFraction);
             var youngestByEvictability = contentHashesWithInfo.SkipOptimized(contentHashesWithInfo.Count / 2).ApproximateSort(comparer, intoEffectiveLastAccessTimes, _configuration.EvictionPoolSize, _configuration.EvictionWindowSize, _configuration.EvictionRemovalFraction);
 
-            return NuCacheCollectionUtilities.MergeOrdered(oldestByEvictability, youngestByEvictability, comparer);
+            return NuCacheCollectionUtilities.MergeOrdered(oldestByEvictability, youngestByEvictability, comparer)
+                .Where((candidate, index) => IsPassEvictionAge(context, candidate, _configuration.EvictionMinAge, ref evictionCount));
+        }
+
+        private bool IsPassEvictionAge(Context context, ContentHashWithLastAccessTimeAndReplicaCount candidate, TimeSpan evictionMinAge, ref int evictionCount)
+        {
+            if (candidate.Age > evictionMinAge)
+            {
+                evictionCount++;
+                return true;
+            }
+
+            context.Debug($"{evictionCount} eviction attempts before current candidate");

```

Comment:
This should be traced in one line.

And I believe that the third line you have right now is pretty stable and I don't think we need to trace it all the time.

Reply:
fixed